### ISSUE TITLE
Very poor performance on One Shot Learning

### ISSUE THREAD

**Issue by chinmay5** [facebookresearch/LASER#78]
Hi, I have been trying to use LASER for product category prediction based on product reviews. The initial dataset is in English and I was able to train it to get an accuracy of 88% using a Siamese Network and 5 Nearest Neighbour. The F1 score for all the categories were reasonable as well.

However, when I used the German dataset for  similar Product Category prediction, the accuracy values are really bad, 8% when there are 22 classes and one can expect ~4% random accuracy. Now, at this point to be honest, I am running out of ideas.

The training is completely in English and the prediction is in German that too Zero Shot. So, I depend completely on the Sentence Embedding. Can anyone give me tips for improving the model since frankly, I am completely confused

**Comment by hoschwenk** [comment]
LASER should perform with zero-shot transfer to a new language, but for the SAME task. 
Reading your text, it seems that the training and transfer task are NOT the same. This is not expected to work (even for English)

**Comment by anassalamah** [comment]
Hello @hoschwenk any pointers on how to achieve the highest accuracy possible with zero-shot transfer to a new language, for the same task of course. I was able to get an 81% on Test in Arabic from training on English with a best dev of 89%. I tried to play around with the size of the training set. I found 18000 elements in train data to be the best (anything more or less gives me less accuracy on Test in Arabic). Some question?

It seems that training for more than around 15 epochs doesn't improve test/train accuracy. Why is that?
Shouldn't more English data produce higher accuracy?
Any general guidelines for applying zero-shot transfer learning on another language?

Thanks

**Comment by PadLex** [comment]
@anassalamah Wait, how are you training LASER? I thought that you couldn't do that without the source code.

**Comment by hoschwenk** [comment]
@PadLex , I believe that "training" refers to the classifier on top of the fixed LASER embeddings

**Comment by hoschwenk** [comment]
You say that training with more data or more than 15 epochs does not improve test/train accuracy.
Is this for English or the zero-shot transfer ?
When your train does not decrease further, this could mean that the capacity of your classifier is not adapted to the size and difficulty of the data, or that the data is contradictory.
81% on tets for Arabic with 89% on dev for English is rather good.
If your zero-shot transfer rate on test is close to the TEST on English, LASER performs a good job :-)
You can't expect to be better in zero-shot than in English
